hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoUpdateHandler.java

```diff
@@ -285,4 +303,115 @@ private void validateFields(
+     * @return the calculated fees
+     */
+    @NonNull
+    @Override
+    public Fees calculateFees(@NonNull final FeeContext feeContext) {
+        // Variable bytes plus two additional longs for balance and auto-renew period; plus a boolean for receiver sig
+        // required.
+        final var body = feeContext.body();
+        final var accountStore = feeContext.readableStore(ReadableAccountStore.class);
+        return cryptoUpdateFees(
+                body, feeContext.feeCalculator(SubType.DEFAULT), accountStore, feeContext.configuration());
+    }
+
+    /**
+     * This method calculates the base size of the cryptoUpdate transaction.
+     * This is the duplicated code as in mono-service
+     * @param op the {@link CryptoUpdateTransactionBody}
+     * @param keySize the size of the key
+     * @return the calculated base size
+     */
+    private static long baseSizeOf(final CryptoUpdateTransactionBody op, final long keySize) {
+        return BASIC_ENTITY_ID_SIZE
+                + op.memoOrElse("").getBytes(StandardCharsets.UTF_8).length
+                + (op.hasExpirationTime() ? LONG_SIZE : 0L)
+                + (op.hasAutoRenewPeriod() ? LONG_SIZE : 0L)
+                + (op.hasProxyAccountID() ? BASIC_ENTITY_ID_SIZE : 0L)
+                + (op.hasMaxAutomaticTokenAssociations() ? INT_SIZE : 0L)
+                + keySize;
+    }
+
+    /**
+     * This method calculates the bytes for the CryptoUpdate transaction auto-renew information.
+     * This is the duplicated code as in mono-service
+     * @param account the {@link Account} to be updated
+     * @return the calculated bytes
+     */
+    private static long cryptoAutoRenewRb(final Account account) {
+        return CRYPTO_ENTITY_SIZES.fixedBytesInAccountRepr()
+                + currentNonBaseBytes(account)
+                + account.numberAssociations() * CRYPTO_ENTITY_SIZES.bytesInTokenAssocRepr();
+    }
+
+    /**
+     * This method calculates the bytes for the CryptoUpdate transaction related to memo and keys.
+     * This is the duplicated code as in mono-service
+     * @param account the {@link Account} to be updated
+     * @return the calculated bytes
+     */
+    private static int currentNonBaseBytes(final Account account) {
+        return account.memo().getBytes(StandardCharsets.UTF_8).length
+                + getAccountKeyStorageSize(fromPbj(account.keyOrElse(Key.DEFAULT)))
+                + (account.maxAutoAssociations() == 0 ? 0 : INT_SIZE);
+    }
+
+    /**
+     * This method calculates the fees for the CryptoUpdate transaction.
+     * This can also be used for lazy account creation logic in AutoAccountCreator class in future PRs
+     * @param body the {@link TransactionBody}
+     * @param feeCalculator the {@link FeeCalculator}
+     * @param accountStore the {@link ReadableAccountStore}
+     * @param configuration the {@link Configuration}
+     * @return the calculated fees
+     */
+    public static Fees cryptoUpdateFees(
```

Comment:
Maybe it is worth to apply `@NonNull` or `Nullable` on the return and params

Reply:
Agreed. Will do in my next PR .Thanks @povolev15